<div style=" background-color: RGB(0,114,200);" >
<h1 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">PROJET 4 DATA ANALYST</h1>
<h2 style="margin: auto; padding: 20px 0; color:#fff; text-align: center">Réalisez une étude de santé publique avec R ou Python
</h2>
</div>

# OBJECTIF DE CE NOTEBOOK

Bienvenue dans l'outil plébiscité par les analystes de données Jupyter.

Il s'agit d'un outil permettant de mixer et d'alterner codes, textes et graphique.

Cet outil est formidable pour plusieurs raisons:

+ il permet de tester des lignes de codes au fur et à mesure de votre rédaction, de constater immédiatement le résultat d'un instruction, de la corriger si nécessaire.
+ De rédiger du texte pour expliquer l'approche suivie ou les résultats d'une analyse et de le mettre en forme grâce à du code html ou plus simple avec **Markdown**
+ d'agrémenter de graphiques

Pour vous aider dans vos premiers pas à l'usage de Jupyter et de Python, nous avons rédigé ce notebook en vous indiquant les instructions à suivre.

Il vous suffit pour cela de saisir le code Python répondant à l'instruction donnée.

Vous verrez de temps à autre le code Python répondant à une instruction donnée mais cela est fait pour vous aider à comprendre la nature du travail qui vous est demandée.

Et garder à l'esprit, qu'il n'y a pas de solution unique pour résoudre un problème et qu'il y a autant de résolutions de problèmes que de développeurs ;)...



Note jeremy 
Est ce qu'il faut faire le calcul de la sous nutrition sur les pays qu'on a ? 
Est ce qu'il faut faire des graphiques ?
Rajouter le soja
La liste des céréales est difficile a trouver ...

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 1 - Importation des librairies et chargement des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.1 - Importation des librairies</h3>
</div>

In [1]:
#Importation de la librairie Pandas
import pandas as pd

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">1.2 - Chargement des fichiers Excel</h3>
</div>

In [2]:
#Importation du fichier population.csv
population = pd.read_csv('population.csv')

#Importation du fichier dispo_alimentaire.csv
dispoAlimentaire = pd.read_csv('dispo_alimentaire.csv')

#Importation du fichier aide_alimentaire.csv
aideAlimentaire = pd.read_csv('aide_alimentaire.csv')

#Importation du fichier sous_nutrition.csv
sousNutrition = pd.read_csv('sous_nutrition.csv')

<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 2 - Analyse exploratoire des fichiers</h2>
</div>

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.1 - Analyse exploratoire du fichier population</h3>
</div>

In [3]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(population.shape[0]))
print("Le tableau comporte {} colonne(s)".format(population.shape[1]))

Le tableau comporte 1416 observation(s) ou article(s)
Le tableau comporte 3 colonne(s)


In [4]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonne(s)".format(len(population.columns)))

#La nature des données dans chacune des colonnes
# Convertir la série des types de données en un DataFrame
dtypes_df = population.dtypes.to_frame()
# Formater le DataFrame en une chaîne
dtypes_str = dtypes_df.to_string()
# Afficher le résultat
print("La nature des colonnes est :\n{}".format(dtypes_str))

#Le nombre de valeurs présentes dans chacune des colonnes
print("Le nombre de valeurs présentes dans chacune des colonnes est :\n{}".format(population.count()))

Le tableau comporte 3 colonne(s)
La nature des colonnes est :
              0
Zone     object
Année     int64
Valeur  float64
Le nombre de valeurs présentes dans chacune des colonnes est :
Zone      1416
Année     1416
Valeur    1416
dtype: int64


Le tableau comporte 3 colonne(s)

La nature des colonnes est :
Zone     object
Année     int64
Valeur  float64

Le nombre de valeurs présentes dans chacune des colonnes est :
Zone      1416
Année     1416
Valeur    1416

In [5]:
#Affichage les 5 premières lignes de la table
print("Les 5 premières lignes de la table population")
print(population.head())


Les 5 premières lignes de la table population
          Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113


Les 5 premières lignes de la table population
          Zone  Année     Valeur
0  Afghanistan   2013  32269.589
1  Afghanistan   2014  33370.794
2  Afghanistan   2015  34413.603
3  Afghanistan   2016  35383.032
4  Afghanistan   2017  36296.113

In [6]:
#Nous allons harmoniser les unités. Pour cela, nous avons décidé de multiplier la population par 1000
#Multiplication de la colonne valeur par 1000

print("Nous allons harmoniser les unités en multipliant la population par 1000.")

# Multiplication de la colonne "Valeur" par 1000
population['Population'] = population['Valeur'] * 1000

# Convertir les valeurs en entiers
population['Population'] = population['Population'].astype(int)

# Afficher le résultat
print("Nous avons harmonisé les unités en multipliant la population par 1000 :\n{}".format(population))


Nous allons harmoniser les unités en multipliant la population par 1000.
Nous avons harmonisé les unités en multipliant la population par 1000 :
             Zone  Année     Valeur  Population
0     Afghanistan   2013  32269.589    32269589
1     Afghanistan   2014  33370.794    33370794
2     Afghanistan   2015  34413.603    34413603
3     Afghanistan   2016  35383.032    35383032
4     Afghanistan   2017  36296.113    36296113
...           ...    ...        ...         ...
1411     Zimbabwe   2014  13586.707    13586707
1412     Zimbabwe   2015  13814.629    13814629
1413     Zimbabwe   2016  14030.331    14030331
1414     Zimbabwe   2017  14236.595    14236595
1415     Zimbabwe   2018  14438.802    14438802

[1416 rows x 4 columns]


Nous avons harmonisé les unités en multipliant la population par 1000 :
0       32269589
1       33370794
2       34413603
3       35383032
4       36296113
          ...   
1411    13586707
1412    13814629
1413    14030331
1414    14236595
1415    14438802

In [7]:
# Supprimer la colonne "Valeur" originale
population.drop(columns=['Valeur'], inplace=True)

# Afficher le résultat
print(population.head())

          Zone  Année  Population
0  Afghanistan   2013    32269589
1  Afghanistan   2014    33370794
2  Afghanistan   2015    34413603
3  Afghanistan   2016    35383032
4  Afghanistan   2017    36296113


   Zone  Année  Population
0  Afghanistan   2013   32269.589
1  Afghanistan   2014   33370.794
2  Afghanistan   2015   34413.603
3  Afghanistan   2016   35383.032
4  Afghanistan   2017   36296.113

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.2 - Analyse exploratoire du fichier disponibilité alimentaire</h3>
</div>

In [8]:
#Afficher les dimensions du dataset

print("Le tableau comporte {} observation(s) ou article(s)".format(dispoAlimentaire.shape[0]))

Le tableau comporte 15605 observation(s) ou article(s)


Le tableau comporte 15605 observation(s) ou article(s)

In [9]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonne(s)".format(dispoAlimentaire.shape[1]))


Le tableau comporte 18 colonne(s)


Le tableau comporte 18 colonne(s)

In [10]:
#Affichage les 5 premières lignes de la table
print("Les 5 premières lignes de la table dispositions alimentaires sont :")
print(dispoAlimentaire.head())


Les 5 premières lignes de la table dispositions alimentaires sont :
          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    NaN   
1  Afghanistan        Agrumes, Autres  vegetale                    NaN   
2  Afghanistan  Aliments pour enfants  vegetale                    NaN   
3  Afghanistan                 Ananas  vegetale                    NaN   
4  Afghanistan                Bananes  vegetale                    NaN   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  NaN                                             5.0   
1                  NaN                                             1.0   
2                  NaN                                             1.0   
3                  NaN                                             0.0   
4                  NaN                                             4.0   

   Disponibilité alimentaire en quantité (

Les 5 premières lignes de la table dispositions alimentaires sont :
          Zone                Produit   Origine  Aliments pour animaux  Autres Utilisations  ...  Pertes  Production  Semences  Traitement  Variation de stock
0  Afghanistan       Abats Comestible   animale                    NaN                  NaN  ...     NaN        53.0       NaN         NaN                 NaN
1  Afghanistan        Agrumes, Autres  vegetale                    NaN                  NaN  ...     2.0         3.0       NaN         NaN                 NaN
2  Afghanistan  Aliments pour enfants  vegetale                    NaN                  NaN  ...     NaN         NaN       NaN         NaN                 NaN
3  Afghanistan                 Ananas  vegetale                    NaN                  NaN  ...     NaN         NaN       NaN         NaN                 NaN
4  Afghanistan                Bananes  vegetale                    NaN                  NaN  ...     NaN         NaN       NaN         NaN                 NaN

In [11]:
#remplacement des NaN dans le dataset par des 0
dispoAlimentaire = dispoAlimentaire.fillna(0)
print(dispoAlimentaire.head())


          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1

     Zone                Produit   Origine  Aliments pour animaux  Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  Disponibilité alimentaire en quantité (kg/personne/an)  ...  Importations - Quantité  Nourriture  Pertes  Production  Semences  Traitement  Variation de stock
0  Afghanistan       Abats Comestible   animale                    0.0                  0.0                                             5.0                                               1.72       ...                      0.0        53.0     0.0        53.0       0.0         0.0                 0.0
1  Afghanistan        Agrumes, Autres  vegetale                    0.0                  0.0                                             1.0                                               1.29       ...                     40.0        39.0     2.0         3.0       0.0         0.0                 0.0
2  Afghanistan  Aliments pour enfants  vegetale                    0.0                  0.0                                             1.0                                               0.06       ...                      2.0         2.0     0.0         0.0       0.0         0.0                 0.0
3  Afghanistan                 Ananas  vegetale                    0.0                  0.0                                             0.0                                               0.00       ...                      0.0         0.0     0.0         0.0       0.0         0.0                 0.0
4  Afghanistan                Bananes  vegetale                    0.0                  0.0                                             4.0                                               2.70       ...                     82.0        82.0     0.0         0.0       0.0         0.0                 0.0

In [12]:
#multiplication de toutes les lignes contenant des milliers de tonnes en Kg
colonnes_a_convertir = ['Autres Utilisations', 'Disponibilité intérieure', 'Exportations - Quantité',
                        'Importations - Quantité', 'Nourriture', 'Pertes', 'Production', 'Semences', 
                        'Traitement', 'Variation de stock']

for colonne in colonnes_a_convertir:
    dispoAlimentaire[colonne] *= 1000  # Convertir de milliers de tonnes à kg



In [13]:
#Affichage les 5 premières lignes de la table
print(dispoAlimentaire.head())

          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1

 Aliments pour animaux  Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  Disponibilité alimentaire en quantité (kg/personne/an)  Disponibilité de matière grasse en quantité (g/personne/jour)  Disponibilité de protéines en quantité (g/personne/jour)  ...  Nourriture   Pertes  Production  Semences  Traitement  Variation de stock
0                      0                    0                                         5000000                                            1720000                                                  200000                                                         770000         ...    53000000        0    53000000         0           0                   0
1                      0                    0                                         1000000                                            1290000                                                   10000                                                          20000         ...    39000000  2000000     3000000         0           0                   0
2                      0                    0                                         1000000                                              60000                                                   10000                                                          30000         ...     2000000        0           0         0           0                   0
3                      0                    0                                               0                                                  0                                                       0                                                              0         ...           0        0           0         0           0                   0
4                      0                    0                                         4000000                                            2700000                                                   20000                                                          50000         ...    82000000        0           0         0           0                   0

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier aide alimentaire</h3>
</div>

In [14]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(aideAlimentaire.shape[0]))
print("Le tableau comporte {} colonne(s)".format(aideAlimentaire.shape[1]))

Le tableau comporte 1475 observation(s) ou article(s)
Le tableau comporte 4 colonne(s)


In [15]:
#Consulter le nombre de colonnes
print("Le nombre de colonnes est : {}".format(aideAlimentaire.shape[1]))


Le nombre de colonnes est : 4


In [16]:
#Affichage les 5 premières lignes de la table
print("les 5 premières lignes de la table aideAlimentaire sont : ")
print(aideAlimentaire.head())

les 5 premières lignes de la table aideAlimentaire sont : 
  Pays bénéficiaire  Année              Produit  Valeur
0       Afghanistan   2013  Autres non-céréales     682
1       Afghanistan   2014  Autres non-céréales     335
2       Afghanistan   2013         Blé et Farin   39224
3       Afghanistan   2014         Blé et Farin   15160
4       Afghanistan   2013             Céréales   40504


In [17]:
#changement du nom de la colonne Pays bénéficiaire par Zone
aideAlimentaire.rename(columns={'Pays bénéficiaire':'Zone'}, inplace=True)
print(aideAlimentaire.head())


          Zone  Année              Produit  Valeur
0  Afghanistan   2013  Autres non-céréales     682
1  Afghanistan   2014  Autres non-céréales     335
2  Afghanistan   2013         Blé et Farin   39224
3  Afghanistan   2014         Blé et Farin   15160
4  Afghanistan   2013             Céréales   40504


In [18]:
#Multiplication de la colonne Aide_alimentaire qui contient des tonnes par 1000 pour avoir des kg
aideAlimentaire['Valeur'] = aideAlimentaire['Valeur']*1000


In [19]:
#Affichage les 5 premières lignes de la table
print(aideAlimentaire.head())

          Zone  Année              Produit    Valeur
0  Afghanistan   2013  Autres non-céréales    682000
1  Afghanistan   2014  Autres non-céréales    335000
2  Afghanistan   2013         Blé et Farin  39224000
3  Afghanistan   2014         Blé et Farin  15160000
4  Afghanistan   2013             Céréales  40504000


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">2.3 - Analyse exploratoire du fichier sous nutrition</h3>
</div>

In [20]:
#Afficher les dimensions du dataset
print("Le tableau comporte {} observation(s) ou article(s)".format(sousNutrition.shape[0]))


Le tableau comporte 1218 observation(s) ou article(s)


In [21]:
#Consulter le nombre de colonnes
print("Le tableau comporte {} colonne(s)".format(sousNutrition.shape[1]))

Le tableau comporte 3 colonne(s)


In [22]:
#Afficher les 5 premières lignes de la table
print(sousNutrition.head())


          Zone      Année Valeur
0  Afghanistan  2012-2014    8.6
1  Afghanistan  2013-2015    8.8
2  Afghanistan  2014-2016    8.9
3  Afghanistan  2015-2017    9.7
4  Afghanistan  2016-2018   10.5


In [23]:
#Conversion de la colonne sous nutrition en numérique
sousNutrition['Valeur'] = pd.to_numeric(sousNutrition['Valeur'], errors='coerce')



In [24]:
#Conversion de la colonne (avec l'argument errors=coerce qui permet de convertir automatiquement les lignes qui ne sont pas des nombres en NaN)
#Puis remplacement des NaN en 0
sousNutrition['Valeur'] = sousNutrition['Valeur'].fillna(0)
print(sousNutrition['Valeur'])


0        8.6
1        8.8
2        8.9
3        9.7
4       10.5
        ... 
1213     0.0
1214     0.0
1215     0.0
1216     0.0
1217     0.0
Name: Valeur, Length: 1218, dtype: float64


In [25]:
#changement du nom de la colonne Valeur par sous_nutrition
sousNutrition = sousNutrition.rename(columns={'Valeur': 'sous_nutrition'})
print(sousNutrition)

             Zone      Année  sous_nutrition
0     Afghanistan  2012-2014             8.6
1     Afghanistan  2013-2015             8.8
2     Afghanistan  2014-2016             8.9
3     Afghanistan  2015-2017             9.7
4     Afghanistan  2016-2018            10.5
...           ...        ...             ...
1213     Zimbabwe  2013-2015             0.0
1214     Zimbabwe  2014-2016             0.0
1215     Zimbabwe  2015-2017             0.0
1216     Zimbabwe  2016-2018             0.0
1217     Zimbabwe  2017-2019             0.0

[1218 rows x 3 columns]


In [26]:
#Multiplication de la colonne sous_nutrition par 1000000
sousNutrition['sous_nutrition'] *= 1000000

# Conversion en chiffres entiers
sousNutrition['sous_nutrition'] = sousNutrition['sous_nutrition'].astype(int)






In [27]:
#Afficher les 5 premières lignes de la table
print(sousNutrition.head())

          Zone      Année  sous_nutrition
0  Afghanistan  2012-2014         8600000
1  Afghanistan  2013-2015         8800000
2  Afghanistan  2014-2016         8900000
3  Afghanistan  2015-2017         9700000
4  Afghanistan  2016-2018        10500000


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.1 - Proportion de personnes en sous nutrition</h3>
</div>

In [28]:
# TRAVAIL SUR LA TABLE SOUS NUTRITION


# Déploiement du groupe d'année sur une ligne ['2012-2014'] => ['2012', '2013', '2014']
sousNutrition['Année'] = sousNutrition['Année'].str.split('-').apply(lambda x: list(range(int(x[0]), int(x[1])+1)))
sousNutrition = sousNutrition.explode('Année')
sousNutrition['Année'] = sousNutrition['Année'].astype(int)

# Réinitialiser les index /garantit que chaque ligne a un index unique et que les index sont contigus, ce qui facilite l'accès et la manipulation des données.
sousNutrition = sousNutrition.reset_index(drop=True)

# Convertir la colonne 'Année' en type entier (int)
sousNutrition['Année'] = sousNutrition['Année'].astype(int)


# ************************************************************


# Jointure entre les tables population et sousNutrition sur les colonnes 'Zone' et 'Année'
jointure = pd.merge(population, sousNutrition, on=['Zone', 'Année'])

# Filtrer les résultats pour ne conserver que les lignes avec l'année 2017
resultats_2017 = jointure[jointure['Année'] == 2017]

#Regrouper les données en fonction de la colonne Zone 
#puis retenir la première valeur de la colonne Population et la dernière valeurs de la colonne sous_nutrition
# les données de la FAO en sous-nutrition sont déjà une moyenne sur trois ans (vérification sur le site FAOSTAT), donc je récupère la valeur la plus récente
# Trouver la valeur de personnes en sous-nutrition pour chaque pays en 2017
resultats_2017 = resultats_2017.groupby('Zone').agg({'Population': 'first', 'sous_nutrition': 'last'}).reset_index()

# Convertir la colonne 'sous_nutrition' en type entier
resultats_2017['sous_nutrition'] = resultats_2017['sous_nutrition'].astype(int)

# Calculer la proportion de sous-nutrition
resultats_2017['proportion_sous_nutrition'] = resultats_2017['sous_nutrition'] / resultats_2017['Population']*100

# Arrondir les valeurs à deux chiffres après la virgule et rajouter le signe % à la fin
resultats_2017['proportion_sous_nutrition'] = resultats_2017['proportion_sous_nutrition'].round(2).astype(str) + '%'



In [29]:
#Affichage du dataset
# Afficher le DataFrame résultant
print(resultats_2017)


                      Zone  Population  sous_nutrition  \
0              Afghanistan    36296113        11100000   
1           Afrique du Sud    57009756         3300000   
2                  Albanie     2884169          100000   
3                  Algérie    41389189         1200000   
4                Allemagne    82658409               0   
..                     ...         ...             ...   
198  États-Unis d'Amérique   325084756               0   
199               Éthiopie   106399924        21500000   
200              Îles Cook       17507               0   
201          Îles Marshall       58058               0   
202           Îles Salomon      636039               0   

    proportion_sous_nutrition  
0                      30.58%  
1                       5.79%  
2                       3.47%  
3                        2.9%  
4                        0.0%  
..                        ...  
198                      0.0%  
199                    20.21%  
200            

In [30]:

# Calcul et affichage du nombre de personnes en état de sous-nutrition en 2017 avec des séparateurs entre les milliers
total_sous_nutrition = round(resultats_2017['sous_nutrition'].sum())

# Utiliser la méthode format avec le spécificateur de format {:,} pour ajouter des séparateurs entre les milliers
formatted_total_sous_nutrition = "{:,.0f}".format(total_sous_nutrition)

print("Le nombre de personnes en état de sous-nutrition en 2017 est de : {} personnes".format(formatted_total_sous_nutrition))

#calcul du nombre de personnes dans la population mondiale en 2017
total_population = round(resultats_2017['Population'].sum())
formatted_total_population = "{:,.0f}".format(total_population)
print("Le nombre de personnes dans la population mondiale en 2017 est de : {} personnes".format(formatted_total_population))

# Calcul et affichage du pourcentage de personnes en état de sous-nutrition dans le monde en 2017
pourcentage_sous_nutrition = total_sous_nutrition / total_population * 100
pourcentage_sous_nutrition_arrondi = round(pourcentage_sous_nutrition, 2)
print("Le pourcentage de personnes en état de sous-nutrition dans le monde en 2017 est de : {}%".format(pourcentage_sous_nutrition_arrondi))


Le nombre de personnes en état de sous-nutrition en 2017 est de : 544,200,000 personnes
Le nombre de personnes dans la population mondiale en 2017 est de : 7,543,798,769 personnes
Le pourcentage de personnes en état de sous-nutrition dans le monde en 2017 est de : 7.21%


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.2 - Nombre théorique de personnes qui pourraient être nourries</h3>
</div>

In [31]:
#Combien mange en moyenne un être humain ? Source => dispoAlimentaire
# Calculer la disponibilité alimentaire moyenne par personne par produit

disponibilite_moyenne_par_personne = dispoAlimentaire[['Produit', 'Disponibilité alimentaire en quantité (kg/personne/an)']].groupby('Produit').mean()
print(disponibilite_moyenne_par_personne)

                        Disponibilité alimentaire en quantité (kg/personne/an)
Produit                                                                       
Abats Comestible                                                 2.730345     
Agrumes, Autres                                                  1.306688     
Alcool, non Comestible                                           0.000000     
Aliments pour enfants                                            0.380936     
Ananas                                                           3.890536     
...                                                                   ...     
Viande de Suides                                                14.407616     
Viande de Volailles                                             20.561322     
Viande, Autre                                                    1.800977     
Vin                                                              6.161337     
Épices, Autres                                      

In [32]:
#Filtrer la table population sur l'année 2017 
population_2017 = population[population['Année'] == 2017]

#Faire une jointure entre les tables disponibilité alimentaire et population sur la colonne Zone
dispoAlimentaire = pd.merge(dispoAlimentaire, population_2017, on='Zone')



In [33]:
#Affichage du nouveau dataframe
# Afficher les premières lignes du DataFrame
print(dispoAlimentaire.head())

          Zone                Produit   Origine  Aliments pour animaux  \
0  Afghanistan       Abats Comestible   animale                    0.0   
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
0                  0.0                                             5.0   
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
0                                               1

In [34]:
# Calculer la somme de la disponibilité alimentaire en calories par personne et par jour par pays
Nbre_cal_journaliere_par_personne_disponibles = dispoAlimentaire.groupby('Zone')['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()
Nbre_cal_journaliere_par_personne_disponibles.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)': 'Nbre_cal_journaliere_par_personne_disponibles'}, inplace=True)

# Fusionner les DataFrames sur la colonne 'Zone'
dispoAlimentaire = pd.merge(dispoAlimentaire, Nbre_cal_journaliere_par_personne_disponibles, on='Zone')

# Grouper par pays et obtenir la première ligne de chaque groupe
NourritureDispo = dispoAlimentaire.groupby('Zone').first().reset_index()[['Zone', 'Population', 'Nbre_cal_journaliere_par_personne_disponibles']]

# Calculer la colonne Nbre_cal_dispo_totale
NourritureDispo['Nbre_cal_dispo_totale'] = NourritureDispo['Population'] * NourritureDispo['Nbre_cal_journaliere_par_personne_disponibles']

# Afficher les premières lignes du DataFrame mis à jour
print(NourritureDispo.head())



             Zone  Population  Nbre_cal_journaliere_par_personne_disponibles  \
0     Afghanistan    36296113                                         2087.0   
1  Afrique du Sud    57009756                                         3020.0   
2         Albanie     2884169                                         3188.0   
3         Algérie    41389189                                         3293.0   
4       Allemagne    82658409                                         3503.0   

   Nbre_cal_dispo_totale  
0           7.574999e+10  
1           1.721695e+11  
2           9.194731e+09  
3           1.362946e+11  
4           2.895524e+11  


In [35]:

# Nombre moyen de calories nécessaires par personne par jour (valeur fictive)
calories_necessaires_par_personne = 2350  # en calories

# Calculer le nombre d'humains pouvant être nourris
NourritureDispo['Nb_humains_nourris'] = NourritureDispo['Nbre_cal_dispo_totale'] / calories_necessaires_par_personne

# Calculer le nombre total d'humains nourris
nombre_total_humains_nourris = NourritureDispo['Nb_humains_nourris'].sum()

print("Le nombre total d'humains pouvant être nourris est d'environ : {:.2f} milliards d'humains".format(nombre_total_humains_nourris / 1e9))




Le nombre total d'humains pouvant être nourris est d'environ : 8.90 milliards d'humains


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.3 - Nombre théorique de personne qui pourrait être nourrie avec les produits végétaux</h3>
</div>

In [36]:
#Transfert des données avec les végétaux dans un nouveau dataframe
dispoAlimentaire_vegetaux = dispoAlimentaire[dispoAlimentaire['Origine'] == 'vegetale']
print(dispoAlimentaire_vegetaux.head())

          Zone                Produit   Origine  Aliments pour animaux  \
1  Afghanistan        Agrumes, Autres  vegetale                    0.0   
2  Afghanistan  Aliments pour enfants  vegetale                    0.0   
3  Afghanistan                 Ananas  vegetale                    0.0   
4  Afghanistan                Bananes  vegetale                    0.0   
6  Afghanistan                  Bière  vegetale                    0.0   

   Autres Utilisations  Disponibilité alimentaire (Kcal/personne/jour)  \
1                  0.0                                             1.0   
2                  0.0                                             1.0   
3                  0.0                                             0.0   
4                  0.0                                             4.0   
6                  0.0                                             0.0   

   Disponibilité alimentaire en quantité (kg/personne/an)  \
1                                               1

In [37]:
#Calcul du nombre de kcal disponible pour les végétaux

# Supprimer la colonne 'Nbre_cal_journaliere_par_personne_disponibles' du DataFrame dispoAlimentaire_vegetaux car elle regroupe toutes les origines
dispoAlimentaire_vegetaux = dispoAlimentaire_vegetaux.drop('Nbre_cal_journaliere_par_personne_disponibles', axis=1)

#calculer le Nbre_cal_vege_journaliere_par_personne_disponibles pour les produits végétaux
Nbre_cal_vege_journaliere_par_personne_disponibles = dispoAlimentaire_vegetaux.groupby('Zone')['Disponibilité alimentaire (Kcal/personne/jour)'].sum().reset_index()
Nbre_cal_vege_journaliere_par_personne_disponibles.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)': 'Nbre_cal_vege_journaliere_par_personne_disponibles'}, inplace=True)

#Ajouter cette colonne au DataFrame dispoAlimentaire_vegetaux
dispoAlimentaire_vegetaux = pd.merge(dispoAlimentaire_vegetaux, Nbre_cal_vege_journaliere_par_personne_disponibles, on='Zone')

#Grouper par pays et obtenir la première ligne de chaque groupe
NourritureDispo_vegetaux = dispoAlimentaire_vegetaux.groupby('Zone').first().reset_index()[['Zone', 'Population', 'Nbre_cal_vege_journaliere_par_personne_disponibles']]
print(NourritureDispo_vegetaux.head())

             Zone  Population  \
0     Afghanistan    36296113   
1  Afrique du Sud    57009756   
2         Albanie     2884169   
3         Algérie    41389189   
4       Allemagne    82658409   

   Nbre_cal_vege_journaliere_par_personne_disponibles  
0                                             1871.0   
1                                             2533.0   
2                                             2203.0   
3                                             2915.0   
4                                             2461.0   


In [38]:
#Calculer la colonne Nbre_cal_vege_dispo_totale
NourritureDispo_vegetaux['Nbre_cal_vege_dispo_totale'] = NourritureDispo_vegetaux['Population'] * NourritureDispo_vegetaux['Nbre_cal_vege_journaliere_par_personne_disponibles']

#Calculer le nombre d'humains pouvant être nourris avec les produits végétaux
NourritureDispo_vegetaux['Nb_humains_nourris_vegetaux'] = NourritureDispo_vegetaux['Nbre_cal_vege_dispo_totale'] / calories_necessaires_par_personne

#Calculer le nombre total d'humains nourris avec les produits végétaux
nombre_total_humains_nourris_vegetaux = NourritureDispo_vegetaux['Nb_humains_nourris_vegetaux'].sum()
print("Le nombre total d'humains pouvant être nourris avec les produits végétaux est d'environ : {:.2f} milliards d'humains".format(nombre_total_humains_nourris_vegetaux / 1e9))

Le nombre total d'humains pouvant être nourris avec les produits végétaux est d'environ : 7.35 milliards d'humains


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.4 - Utilisation de la disponibilité intérieure</h3>
</div>

In [39]:
#Calcul de la disponibilité totale
dispo_int = dispoAlimentaire['Disponibilité intérieure'].sum()
print("La disponibilité intérieure mondiale est de : {:.2f} kilos".format(dispo_int))
print("La disponibilité intérieure mondiale est de : {:.2f} tonnes".format(dispo_int/1000))


La disponibilité intérieure mondiale est de : 9733927000.00 kilos
La disponibilité intérieure mondiale est de : 9733927.00 tonnes


In [40]:
#création d'une boucle for pour afficher les différentes valeurs en fonction des colonnes aliments pour animaux, pertes, nourritures, 
colonnes = ['Aliments pour animaux', 'Pertes', 'Nourriture']
for colonne in colonnes:
    dispo_int_colonne = dispoAlimentaire[colonne].sum()
    print("La disponibilité intérieure mondiale en {} est de : {:.2f} kilos".format(colonne, dispo_int_colonne))


La disponibilité intérieure mondiale en Aliments pour animaux est de : 1288002.00 kilos
La disponibilité intérieure mondiale en Pertes est de : 452283000.00 kilos
La disponibilité intérieure mondiale en Nourriture est de : 4805525000.00 kilos


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.5 - Utilisation des céréales</h3>
</div>

In [41]:
#Création d'une liste avec toutes les variables du fichier DispoAlimentaire
variables = ['Autres Utilisations', 
             'Aliments pour animaux',
             'Nourriture', 
             'Pertes', 
             'Semences', 
             'Traitement',
             'Exportations - Quantité',
             'Importations - Quantité',
             'Production',
             'Variation de stock',
             'Disponibilité intérieure'
             ]


In [42]:
#création d'un dataframe qui filtre sur toutes les céréales ['Blé', 'Riz (Eq Blanchi)', 'Orge', 'Maïs', 'Millet', 'Seigle', 'Avoine', 'Sorgho', 'Céréales, Autres']
cereales = dispoAlimentaire[dispoAlimentaire['Produit'].isin(['Blé', 'Riz (Eq Blanchi)', 'Orge', 'Maïs', 'Millet', 'Seigle', 'Avoine', 'Sorgho', 'Céréales, Autres'])]


In [43]:
# Calculer la proportion totale de 'Aliments pour animaux' par rapport à la disponibilité intérieure totale
proportion_aliments_animaux = cereales['Aliments pour animaux'].sum() / cereales['Disponibilité intérieure'].sum() * 100
print("La proportion totale d'Aliments pour animaux par rapport à la disponibilité intérieure totale est de : {:.2f}%".format(proportion_aliments_animaux))



La proportion totale d'Aliments pour animaux par rapport à la disponibilité intérieure totale est de : 0.04%


In [44]:

# Calculer la proportion totale de 'Nourriture'humaine par rapport à la disponibilité intérieure totale
proportion_nourriture = cereales['Nourriture'].sum() / cereales['Disponibilité intérieure'].sum() * 100
print("La proportion totale de Nourriture humaine par rapport à la disponibilité intérieure totale est de : {:.2f}%".format(proportion_nourriture))

La proportion totale de Nourriture humaine par rapport à la disponibilité intérieure totale est de : 42.91%


In [45]:

# Calculer la proportion totale par variable par rapport à la disponibilité intérieure totale
proportions = cereales[variables].sum() / cereales['Disponibilité intérieure'].sum() * 100
print(proportions)
print("La différence entre les (importations et les exportations) + la variation de stock  annule le surplus de production de céréales")

Autres Utilisations           9.782830
Aliments pour animaux         0.036143
Nourriture                   42.906006
Pertes                        4.486516
Semences                      2.847285
Traitement                    3.858986
Exportations - Quantité      17.636483
Importations - Quantité      16.461814
Production                  104.934974
Variation de stock           -3.760053
Disponibilité intérieure    100.000000
dtype: float64
La différence entre les (importations et les exportations) + la variation de stock  annule le surplus de production de céréales


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.6 - Pays avec la proportion de personnes sous-alimentée la plus forte en 2017</h3>
</div>

In [46]:
# Trier les données par ordre décroissant de proportion de sous-nutrition
pays_plus_sous_alimentes = resultats_2017.sort_values(by='proportion_sous_nutrition', ascending=False)

# Obtenir le pays avec la proportion la plus élevée
pays_max_sous_alimentes = pays_plus_sous_alimentes.iloc[0]

# Afficher les résultats
print("Pays avec la proportion de personnes sous-alimentées la plus élevée en 2017 :")
print("Pays :", pays_max_sous_alimentes['Zone'])
print("Proportion de sous-nutrition :", pays_max_sous_alimentes['proportion_sous_nutrition'])

Pays avec la proportion de personnes sous-alimentées la plus élevée en 2017 :
Pays : Sénégal
Proportion de sous-nutrition : 9.73%


In [47]:

# Trier les données par ordre décroissant de proportion de sous-nutrition
top_10_pays_sous_alimentes = resultats_2017.sort_values(by='proportion_sous_nutrition', ascending=False).head(10)

# Afficher les résultats
print("Les 10 pays avec la proportion de personnes sous-alimentées la plus élevée en 2017 :")
for index, row in top_10_pays_sous_alimentes.iterrows():
    print("Pays :", row['Zone'])
    print("Proportion de sous-nutrition :", row['proportion_sous_nutrition'])
    print("-----------------------------")

Les 10 pays avec la proportion de personnes sous-alimentées la plus élevée en 2017 :
Pays : Sénégal
Proportion de sous-nutrition : 9.73%
-----------------------------
Pays : El Salvador
Proportion de sous-nutrition : 9.39%
-----------------------------
Pays : Thaïlande
Proportion de sous-nutrition : 9.39%
-----------------------------
Pays : Jordanie
Proportion de sous-nutrition : 9.2%
-----------------------------
Pays : Indonésie
Proportion de sous-nutrition : 9.11%
-----------------------------
Pays : Équateur
Proportion de sous-nutrition : 8.94%
-----------------------------
Pays : Paraguay
Proportion de sous-nutrition : 8.74%
-----------------------------
Pays : Oman
Proportion de sous-nutrition : 8.57%
-----------------------------
Pays : Sri Lanka
Proportion de sous-nutrition : 7.57%
-----------------------------
Pays : Géorgie
Proportion de sous-nutrition : 7.48%
-----------------------------


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.7 - Pays qui ont le plus bénéficié d'aide alimentaire depuis 2013</h3>
</div>

In [48]:
#calcul du total de l'aide alimentaire par pays
# Somme de l'aide alimentaire par pays depuis 2013
aide_total_par_pays = aideAlimentaire.groupby('Zone')['Valeur'].sum().reset_index()


# Trier les pays par ordre décroissant selon le montant total d'aide alimentaire reçue
top_10_pays_aide = aide_total_par_pays.sort_values(by='Valeur', ascending=False).head(11)



In [49]:
#affichage après tri des 10 pays qui ont bénéficié le plus de l'aide alimentaire
# Afficher les résultats
print("Les 10 pays qui ont reçu le plus d'aide alimentaire depuis 2013 sont :")
for index, row in top_10_pays_aide.iterrows():
    print("Pays :", row['Zone'])
    print("Montant total de l'aide alimentaire :", row['Valeur'])
    print("-----------------------------")

Les 10 pays qui ont reçu le plus d'aide alimentaire depuis 2013 sont :
Pays : République arabe syrienne
Montant total de l'aide alimentaire : 1858943000
-----------------------------
Pays : Éthiopie
Montant total de l'aide alimentaire : 1381294000
-----------------------------
Pays : Yémen
Montant total de l'aide alimentaire : 1206484000
-----------------------------
Pays : Soudan du Sud
Montant total de l'aide alimentaire : 695248000
-----------------------------
Pays : Soudan
Montant total de l'aide alimentaire : 669784000
-----------------------------
Pays : Kenya
Montant total de l'aide alimentaire : 552836000
-----------------------------
Pays : Bangladesh
Montant total de l'aide alimentaire : 348188000
-----------------------------
Pays : Somalie
Montant total de l'aide alimentaire : 292678000
-----------------------------
Pays : République démocratique du Congo
Montant total de l'aide alimentaire : 288502000
-----------------------------
Pays : Niger
Montant total de l'aide alim

<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.8 - Evolution des 5 pays qui ont le plus bénéficiés de l'aide alimentaire entre 2013 et 2016</h3>
</div>

In [50]:
#Création d'un dataframe avec la zone, l'année et l'aide alimentaire puis groupby sur zone et année 
# Filtrer les données pour les années 2013 à 2016
aide_2013_2016 = aideAlimentaire[(aideAlimentaire['Année'] >= 2013) & (aideAlimentaire['Année'] <= 2016)]


# Calculer le total de l'aide alimentaire par pays pour la période de 2013 à 2016
aide_total_par_pays = aide_2013_2016.groupby('Zone')['Valeur'].sum().reset_index()



In [51]:
# Trier les pays par ordre décroissant selon le montant total d'aide alimentaire reçue
top_5_pays_aide = aide_total_par_pays.nlargest(5, 'Valeur')
# Afficher les résultats
print("Les 5 pays qui ont reçu le plus d'aide alimentaire entre 2013 et 2016 sont :")
print(top_5_pays_aide[['Zone', 'Valeur']])

Les 5 pays qui ont reçu le plus d'aide alimentaire entre 2013 et 2016 sont :
                         Zone      Valeur
50  République arabe syrienne  1858943000
75                   Éthiopie  1381294000
70                      Yémen  1206484000
61              Soudan du Sud   695248000
60                     Soudan   669784000


In [52]:
# Liste des 5 pays qui ont reçu le plus d'aide alimentaire entre 2013 et 2016
top_5_pays = top_5_pays_aide['Zone'].tolist()

# Filtrer les données pour inclure uniquement les 5 pays dans la période de 2013 à 2016
aide_top_5_pays = aideAlimentaire[(aideAlimentaire['Année'] >= 2013) & (aideAlimentaire['Année'] <= 2016) & (aideAlimentaire['Zone'].isin(top_5_pays))]

# Regrouper les données par année et par pays pour calculer le total de l'aide alimentaire reçue chaque année par ces pays
aide_total_par_annee_pays = aide_top_5_pays.groupby(['Année', 'Zone'])['Valeur'].sum().unstack()

# Remplacer les NaN par 0 dans l'ensemble du DataFrame aide_total_par_annee_pays
aide_total_par_annee_pays = aide_total_par_annee_pays.fillna(0)
print(aide_total_par_annee_pays)

Zone   République arabe syrienne       Soudan  Soudan du Sud        Yémen  \
Année                                                                       
2013                 563566000.0  330230000.0    196330000.0  264764000.0   
2014                 651870000.0  321904000.0    450610000.0  103840000.0   
2015                 524949000.0   17650000.0     48308000.0  372306000.0   
2016                 118558000.0          0.0            0.0  465574000.0   

Zone      Éthiopie  
Année               
2013   591404000.0  
2014   586624000.0  
2015   203266000.0  
2016           0.0  


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.9 - Pays avec le moins de disponibilité par habitant</h3>
</div>

In [53]:
# Calculer la somme totale de Disponibilité alimentaire en quantité par pays
dispo_total_par_pays = dispoAlimentaire.groupby('Zone')['Disponibilité alimentaire en quantité (kg/personne/an)'].sum().reset_index()

# Trier les pays par ordre croissant selon leur Disponibilité alimentaire en quantité
pays_dispo_faible = dispo_total_par_pays.sort_values(by='Disponibilité alimentaire en quantité (kg/personne/an)')


In [54]:
# Afficher les 10 pays ayant la Disponibilité alimentaire en quantité la plus faible
print("Les 10 pays ayant la Disponibilité alimentaire en quantité la plus faible sont :")
print(pays_dispo_faible.head(10))


Les 10 pays ayant la Disponibilité alimentaire en quantité la plus faible sont :
              Zone  Disponibilité alimentaire en quantité (kg/personne/an)
150          Tchad                                             319.45     
165       Zimbabwe                                             336.59     
164         Zambie                                             350.12     
0      Afghanistan                                             351.41     
170       Éthiopie                                             354.55     
62   Guinée-Bissau                                             358.48     
14      Bangladesh                                             364.54     
152    Timor-Leste                                             369.10     
163          Yémen                                             375.41     
148        Sénégal                                             376.72     


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.10 - Pays avec le plus de disponibilité par habitant</h3>
</div>

In [55]:
# Trier les pays par ordre décroissant selon la Disponibilité alimentaire en quantité
pays_dispo_elevee = dispo_total_par_pays.sort_values(by='Disponibilité alimentaire en quantité (kg/personne/an)', ascending=False)

# Afficher les 10 premiers pays ayant la Disponibilité alimentaire en quantité la plus haute
print("Les 10 pays ayant la Disponibilité alimentaire en quantité la plus haute sont :")
print(pays_dispo_elevee.head(10))


Les 10 pays ayant la Disponibilité alimentaire en quantité la plus haute sont :
           Zone  Disponibilité alimentaire en quantité (kg/personne/an)
102  Monténégro                                            1210.41     
89   Luxembourg                                            1150.66     
72      Irlande                                            1150.15     
2       Albanie                                            1095.54     
52     Finlande                                            1093.77     
59        Grèce                                            1070.90     
119    Pays-Bas                                            1069.22     
11     Autriche                                            1069.20     
88     Lituanie                                            1050.07     
44     Danemark                                            1031.04     


<div style="border: 1px solid RGB(0,150,250);" >
<h3 style="margin: auto; padding: 20px; color: RGB(0,150,250); ">3.11 - Exemple de la Thaïlande pour le Manioc</h3>
</div>

In [56]:

#création d'un dataframe avec uniquement la Thaïlande 
# Filtrer les résultats pour ne conserver que les lignes avec l'année 2017 et la Thaïlande
resultats_thailande_2017 = resultats_2017[resultats_2017['Zone'] == 'Thaïlande']
print(resultats_thailande_2017)

          Zone  Population  sous_nutrition proportion_sous_nutrition
176  Thaïlande    69209810         6500000                     9.39%


In [57]:
# Calculer le nombre de personnes en état de sous-nutrition en Thaïlande en 2017
sous_nutrition_thailande_2017 = resultats_thailande_2017['sous_nutrition'].sum()
print("Le nombre de personnes en état de sous-nutrition en Thaïlande en 2017 est de : {:,.0f} personnes".format(sous_nutrition_thailande_2017))

# Calculer le nombre de personnes dans la population en Thaïlande en 2017
population_thailande_2017 = population[(population['Zone'] == 'Thaïlande') & (population['Année'] == 2017)]['Population'].values[0]
print("Le nombre de personnes dans la population en Thaïlande en 2017 est de : {:,.0f} personnes".format(population_thailande_2017))

# Calculer le pourcentage de sous-nutrition en Thaïlande en 2017
pourcentage_sous_nutrition_thailande = (sous_nutrition_thailande_2017 / population_thailande_2017) * 100
print(f"Le pourcentage de sous-nutrition en Thaïlande en 2017 est de : {pourcentage_sous_nutrition_thailande:.2f}%")


Le nombre de personnes en état de sous-nutrition en Thaïlande en 2017 est de : 6,500,000 personnes
Le nombre de personnes dans la population en Thaïlande en 2017 est de : 69,209,810 personnes
Le pourcentage de sous-nutrition en Thaïlande en 2017 est de : 9.39%


In [58]:
# Filtrer les données pour ne conserver que celles concernant le Manioc en Thaïlande
manioc_thailande_data = dispoAlimentaire[(dispoAlimentaire['Zone'] == 'Thaïlande') & (dispoAlimentaire['Produit'] == 'Manioc')]

# Calculer la quantité de Manioc Thaïlandais exporté
exportations_manioc_thailande = manioc_thailande_data['Exportations - Quantité'].sum()


#Calculer la production de Manioc en Thaïlande
production_manioc_thailande = manioc_thailande_data['Production'].sum()

# Calculer la proportion d'exportations par rapport à la production
proportion_export_production = exportations_manioc_thailande / production_manioc_thailande

print(f"Proportion d'exportations de Manioc par rapport à la production : {proportion_export_production:.2%}")


Proportion d'exportations de Manioc par rapport à la production : 83.41%


In [59]:
#Quelle est la disponibilité alimentaire en manioc pour la Thaïlande
# Filtrer les données pour ne conserver que celles concernant la Thaïlande et le manioc
dispo_manioc_thailande = dispoAlimentaire[(dispoAlimentaire['Zone'] == 'Thaïlande') & (dispoAlimentaire['Produit'] == 'Manioc')]
# Calculer la disponibilité par personne de manioc en Thaïlande (kg)
dispo_totale_manioc_thailande = dispo_manioc_thailande['Disponibilité alimentaire en quantité (kg/personne/an)'].sum()
print(f"La disponibilité alimentaire en manioc (kg/personne/an) en Thaïlande est de : {dispo_totale_manioc_thailande:.2f} kg")


La disponibilité alimentaire en manioc (kg/personne/an) en Thaïlande est de : 13.00 kg


<div style="background-color: RGB(0,150,250);" >
<h2 style="margin: auto; padding: 20px; color:#fff; ">Etape 6 - Analyse complémentaires</h2>
</div>